lib/jsonapi-serializers/serializer.rb

```diff
@@ -364,6 +364,15 @@ def self.find_recursive_relationships(root_object, root_inclusion_tree, results)
           is_collection = true
           object = serializer.has_many_relationships[unformatted_attr_name]
         end
+
+        if is_valid_attr && attribute_name.include?("_")
+          expected_name = serializer.format_name(attribute_name)
```

Comment:
This isn't safe because `format_name` should only be called if the attribute name is guaranteed to exist (it'll probably raise a NoMethodError if not) — so if I pass `include: ['invalid_attr']` this will die incorrectly.

Just do this instead:
`expected_name = attribute_name.gsub('_', '-')`